src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,361 @@
+	m_cur_floppy(nullptr),
+	m_bank_select(0),
+	m_side(0)
+{ }
+
+a2bus_superdrive_device::a2bus_superdrive_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock):
+	a2bus_superdrive_device(mconfig, A2BUS_SUPERDRIVE, tag, owner, clock)
+{ }
+
+void a2bus_superdrive_device::device_start()
+{
+	save_item(NAME(m_bank_select));
+	save_item(NAME(m_side));
+}
+
+
+void a2bus_superdrive_device::device_reset()
+{
+	m_bank_select = 0;
+	m_side = 0;
+	m_65c02->reset();
+}
+
+
+// overrides of standard a2bus slot functions
+uint8_t a2bus_superdrive_device::read_c0nx(uint8_t offset)
+{
+	if (machine().side_effects_disabled()) return 0;
+
+	m_bank_select = offset & 0x0f;
+	return 0;
+}
+
+void a2bus_superdrive_device::write_c0nx(uint8_t offset, uint8_t data)
+{
+	if (machine().side_effects_disabled()) return;
+
+	m_bank_select = offset & 0x0f;
+}
+
+uint8_t a2bus_superdrive_device::read_cnxx(uint8_t offset)
+{
+	return m_ram[0x7b00 + offset];
+}
+
+void a2bus_superdrive_device::write_cnxx(uint8_t offset, uint8_t data)
+{
+	//m_ram[0x7b00 + offset] = data;
+}
+
+/*
+ * $c800 - $cbff is uc RAM, controlled via c0nx
+ * $cc00 - $cfff is uc RAM, hardcoded to $7c00 - $7fff
+ */
+uint8_t a2bus_superdrive_device::read_c800(uint16_t offset)
+{
+
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	return m_ram[address];
+}
+
+void a2bus_superdrive_device::write_c800(uint16_t offset, uint8_t data)
+{
+
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	m_ram[address] = data;
+}
+
+
+/* uc 65c02 i/o at $0a00 */
+void a2bus_superdrive_device::m65c02_w(offs_t offset, uint8_t value)
+{
+	if (machine().side_effects_disabled()) return;
+
+	// $00-$0f = swim registers
+	// $10 
+	// $80 = diagnostic led on
+	// $81 = diagnostic led off
+
+	// $40 = head sel low
+	// $41 = head sel high
+
+	if (offset < 16)
+	{
+		m_fdc->write(offset, value);
+		return;
+	}
+
+	if (offset == 0x40)
+	{
+		if (m_cur_floppy) m_cur_floppy->ss_w(0);
+		return;
+	}
+
+	if (offset == 0x41)
+	{
+		if (m_cur_floppy) m_cur_floppy->ss_w(1);
+		return;
+	}
+
+	if (offset == 0x80)
+	{
+		logerror("LED on");
+		return;
+	}
+
+	if (offset == 0x81)
+	{
+		logerror("LED off");
+		return;
+	}
+
+
+	logerror("write($0a%02x,%02x)\n", offset, value);
+}
+
+uint8_t a2bus_superdrive_device::m65c02_r(offs_t offset) {
+	if (machine().side_effects_disabled()) return 0;
+
+
+	if (offset < 16)
+	{
+		return m_fdc->read(offset);
+	}
+
+	if (offset == 0x40)
+	{
+		m_side = 0;
+		if (m_cur_floppy) m_cur_floppy->ss_w(0);
+		return 0;
+	}
+
+	if (offset == 0x41)
+	{
+		m_side = 1;
+		if (m_cur_floppy) m_cur_floppy->ss_w(1);
+		return 0;
+	}
+
+	if (offset == 0x80) {
+		logerror("LED on");
+		return 0;
+	}
+
+	if (offset == 0x81) {
+		logerror("LED off");
+		return 0;
+	}
+
+	logerror("read($0a%02x)\n", offset);
```

Comment:
This whole thing would make more sense as a switch with this logerror in the `default` case and the `return` after it falls off the end, to make it clear it catches specific addresses and logs unhandled addresses.